core/src/main/kotlin/io/customer/android/core/di/DiGraph.kt

```diff
@@ -1,48 +1,111 @@
 package io.customer.android.core.di
 
 abstract class DiGraph {
+    /**
+     * Map of dependencies that can be overridden with mocks in test functions.
+     */
     val overrides: MutableMap<String, Any> = mutableMapOf()
 
     /**
-     * Call when you want to override a dependency with a mock in test functions.
-     * `di.overrideDependency(CustomerIOInstance::class.java, customerIOMock)`
-     *
-     * See `CustomerIOTest` (or any other class that is top-level of the module) for an example on use.
+     * Map of singleton instances of dependencies in the graph.
+     */
+    val singletons: MutableMap<String, Any> = mutableMapOf()
+
+    /**
+     * Internal function to get an overridden dependency in the graph.
+     * This function is marked as public only because it is inlined in other public functions.
+     * Do not call this function directly. Instead, use the `newInstance` or `singleton` functions
+     * to get the dependency.
      */
-    fun <DEP> overrideDependency(dependency: Class<DEP>, value: DEP) {
-        overrides[dependency.simpleName] = value as Any
+    inline fun <reified Dependency : Any> getOverriddenInstance(): Dependency? {
+        return overrides[Dependency::class.java.name] as? Dependency
```

Comment:
how do you feel about adding a method such as 

```
    private inline fun <reified Dependency : Any> dependencyKey(): String = Dependency::class.java.name

```

so we don't have to keep writing the whole `Dependency::class.java.name` thing over and over again ? :D 

usage could be something like

```
inline fun <reified Dependency : Any> getOverriddenInstance(): Dependency? {
        return overrides[dependencyKey<Dependency>()] as? Dependency
    }
```

Reply:
Good idea, added this function.